src/builder/BridgeRoutes.sol

```diff
@@ -0,0 +1,465 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+import {TransferActions} from "../DeFiScripts.sol";
+
+contract QuarkBuilder {
+    /* ===== Constants ===== */
+    string constant VERSION = "1.0.0";
+
+    string constant PAYMENT_METHOD_OFFCHAIN = "OFFCHAIN";
+    string constant PAYMENT_METHOD_PAYCALL = "PAY_CALL";
+    string constant PAYMENT_METHOD_QUOTECALL = "QUOTE_CALL";
+
+    string constant ACTION_TYPE_BRIDGE = "BRIDGE";
+    string constant ACTION_TYPE_TRANSFER = "TRANSFER";
+
+    string constant PAYMENT_CURRENCY_USD = "usd";
+    string constant PAYMENT_CURRENCY_USDC = "usdc";
+
+    /* ===== Custom Errors ===== */
+
+    error AssetPositionNotFound();
+    error BridgeNotFound(uint256 srcChainId, uint256 dstChainId, string assetSymbol)
+    error FundsUnavailable();
+    error InsufficientFunds();
+    error InvalidInput();
+    error MaxCostTooHigh();
+
+    /* ===== Input Types ===== */
+
+    struct ChainAccounts {
+        uint256 chainId;
+        QuarkState[] quarkStates;
+        AssetPositions[] assetPositionsList;
+    }
+
+    // We map this to the Portfolio data structure that the client will already have.
+    // This includes fields that builder may not necessarily need, however it makes
+    // the client encoding that much simpler.
+    struct QuarkState {
+        address account;
+        bool hasCode;
+        bool isQuark;
+        string quarkVersion;
+        uint256 quarkNextNonce;
+    }
+
+    // Similarly, this is designed to intentionally reduce the encoding burden for the client
+    // by making it equivalent in structure to data already in portfolios.
+    struct AssetPositions {
+        address asset;
+        string symbol;
+        uint256 decimals;
+        uint256 usdPrice;
+        AccountBalance[] accountBalances;
+    }
+
+    struct AccountBalance {
+        address account;
+        uint256 balance;
+    }
+
+    struct Payment {
+        // TODO: rename?
+        bool isToken;
+        string currency;
+        // TODO: make into struct?
+        uint256[] chainIds;
+        uint256[] maxCosts;
```

Comment:
may as well, simply for the sake of consistency.

Reply:
Done. Created a `PaymentMaxCost` struct.